usecases/auth/authorization/rbac/manager.go

```diff
@@ -14,19 +14,31 @@ package rbac
 import (
 	"fmt"
 
-	"github.com/casbin/casbin/v2"
 	"github.com/sirupsen/logrus"
 	"github.com/weaviate/weaviate/entities/models"
-	"github.com/weaviate/weaviate/usecases/auth/authorization"
 	"github.com/weaviate/weaviate/usecases/auth/authorization/errors"
 )
 
 type manager struct {
-	casbin *casbin.SyncedCachedEnforcer
+	casbin enforcer
 	logger logrus.FieldLogger
 }
 
-func New(casbin *casbin.SyncedCachedEnforcer, logger logrus.FieldLogger) *manager {
+type enforcer interface {
+	AddNamedPolicy(ptype string, params ...any) (bool, error)
+	GetNamedPolicy(ptype string) ([][]string, error)
+	GetFilteredNamedPolicy(ptype string, fieldIndex int, fieldValues ...string) ([][]string, error)
+	RemoveFilteredNamedPolicy(ptype string, fieldIndex int, fieldValues ...string) (bool, error)
+	SavePolicy() error
+	InvalidateCache() error
+	AddRoleForUser(user string, role string, domains ...string) (bool, error)
+	GetRolesForUser(user string, domains ...string) ([]string, error)
+	DeleteRoleForUser(user string, role string, domains ...string) (bool, error)
+	GetUsersForRole(role string, domains ...string) ([]string, error)
+	Enforce(rvals ...interface{}) (bool, error)
+}
+
+func New(casbin enforcer, logger logrus.FieldLogger) *manager {
 	return &manager{casbin, logger}
 }
```

Comment:
**suggestion:**  i think it's not needed it duplicates  `authorization.Controller` , so we can use it directly

Reply:
This interface is inside `RbacManager` so that `manager_test.go` can mock the `casbin` dependency away and so unit test the logic of its methods in isolation